File: backend/app/schemas/category.py
import uuid
from datetime import datetime
from pydantic import BaseModel, Field


class CategoryBase(BaseModel):
    name: str = Field(..., max_length=100)
    color: str = Field(default="#6366f1", pattern=r"^#[0-9a-fA-F]{6}$")
    keywords: list[str] = Field(default_factory=list)
    prompt: str | None = None


class CategoryCreate(CategoryBase):
    pass


class CategoryUpdate(BaseModel):
    name: str | None = Field(None, max_length=100)
    color: str | None = Field(None, pattern=r"^#[0-9a-fA-F]{6}$")
    keywords: list[str] | None = None
    prompt: str | None = None


class CategoryWeightOut(BaseModel):
    weight: float
    manual_weight: float = 1.0
    total_marked: int

    model_config = {"from_attributes": True}


class CategoryOut(CategoryBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    weight: CategoryWeightOut | None = None
    item_count: int = 0

    model_config = {"from_attributes": True}

File: frontend/src/types/category.ts
export interface CategoryWeight {
  weight: number;
  manual_weight: number;
  total_marked: number;
}

export interface Category {
  id: string;
  name: string;
  color: string;
  keywords: string[];
  prompt: string | null;
  taxonomy_id: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  weight: CategoryWeight | null;
  item_count: number;
}

export interface CategoryCreate {
  name: string;
  color: string;
  keywords: string[];
  prompt?: string | null;
  taxonomy_id?: string | null;
}

export interface CategoryUpdate {
  name?: string;
  color?: string;
  keywords?: string[];
  prompt?: string | null;
  is_active?: boolean;
}
